shared/src/commonMain/kotlin/org/hyperskill/app/step_completion/presentation/StepCompletionActionDispatcher.kt

```diff
@@ -267,4 +279,34 @@ class StepCompletionActionDispatcher(
             .getState(forceUpdate = true)
             .map { it.gamification.hypercoinsBalance }
             .getOrNull()
+
+    private suspend fun handleFetchNextInterviewStep(
+        onNewMessage: (Message) -> Unit
+    ) {
+        sentryInteractor.withTransaction(
+            HyperskillSentryTransactionBuilder.buildStepCompletionNextInterviewStepLoading(),
+            onError = {
+                InternalMessage.FetchNextInterviewStepResultError(
+                    resourceProvider.getString(
+                        SharedResources.strings.step_theory_failed_to_continue_practicing
+                    )
+                )
+            }
+        ) {
+            val nextInterviewStepId = interviewStepsStateRepository
+                .getState(forceUpdate = false)
+                .getOrThrow()
+                .shuffled()
+                .firstOrNull()
```

Comment:
Check shared task description:

1. We should shuffle state repository on widget click
2. Get last step

Reply:
Honestly, I don't see the difference where to shuffle steps and what the step to take (first, or last).
Anyway, I'll move the shuffling out from the feature, to not modify state in the feature and in the repository (as you mentioned before).